### PR TITLE
pull request ready

### PR DESCRIPTION
**Pull request by andypasso**

### CODE REVIEW

File: enumerables.rb [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    my_each { |i| return true if yield i }
+    false
+  end
+
+  def my_none
+    my_each { |i| return true unless yield i }
+    false
+  end
+
+  def my_count
```
Comment (bolah2009) [inline enumerables.rb]: #### `REQUIRED`

### :bug: Let's add a case for when a block is not given. Compare with the built-in method. Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-count)

```ruby
$ irb
irb(main):001:0> require_relative 'enumerables'
=> true
irb(main):002:0> [1,2,3,4,5].count
=> 5
irb(main):003:0> [1,2,3,4,5].my_count
Traceback (most recent call last):
        7: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        6: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        5: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        4: from (irb):3
        3: from /../enumerables.rb:43:in `my_count'
        2: from /../enumerables.rb:7:in `my_each'
        1: from /../enumerables/enumerables.rb:43:in `block in my_count'
LocalJumpError (no block given (yield))
```

File: enumerables.rb [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    count = 0
+    my_each { |i| count += 1 if yield i }
+    count
+  end
+
+  #   def my_map
+  #     array = []
+  #     my_each { |i| array.push yield i }
+  #   end
```
Comment (bolah2009) [inline enumerables.rb]: #### `REQUIRED`

Consider deleting commented out codes. If you need to review a previous version of your code, use git history.

File: enumerables.rb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+# test2 = test.map(&:upcase)
+# # puts test2
+
+# puts [3, 6, 10].my_inject(10) {|sum, number| sum + number}
+
+# puts [3, 6, 10].my_inject(1) {|sum, number| sum * number}
+
+# puts [3, 6, 10, 13].my_inject(0, :+)
+
+# puts ([3, 6, 10].multiply_els { |sum, number| sum * number })
```
Comment (bolah2009) [inline enumerables.rb:94]: Consider deleting commented out codes.

File: enumerables.rb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  #     array = []
+  #     my_each { |i| array.push yield i }
+  #   end
+  def my_map(&procs)
+    array = []
+    my_each { |n| array << (procs.nil? ? yield(n) : procs.call(n)) }
+    array
+  end
+
+  def my_inject(method)
```
Comment (bolah2009) [inline enumerables.rb]: #### `REQUIRED`

### :bug: Let's add a case for when a block is given. Compare with the built-in method. Learn more [here](https://ruby-doc.org/core-2.6.3/Enumerable.html#method-i-inject)

```ruby
irb(main):004:0> [1,2,3,4,5].inject{ |sum, n| sum + n }
=> 15
irb(main):005:0> [1,2,3,4,5].my_inject{ |sum, n| sum + n }
Traceback (most recent call last):
        5: from /../.rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        4: from /../.rbenv/versions/2.6.3/bin/irb:23:in `load'
        3: from /../.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):5
        1: from /../enumerables.rb:57:in `my_inject'
ArgumentError (wrong number of arguments (given 0, expected 1))
```

**Review by bolah2009 — CHANGES_REQUESTED** [review]
# Hi @andynarf  :wave: 

### Awesome Job :muscle: 

Please, test all methods with the built-in methods as shown in the comment below, currently, two methods (`my_count` and `my_inject`) are not working properly.

You can find expected behaviors of these methods [here](https://ruby-doc.org/core-2.6.3/Enumerable.html)

Fix this and submit another code review request. :ok_hand: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

**Review by bellom — APPROVED** [review]
Hi @andynarf,
**Great work making all the necessary corrections from the previous review.**

**NOTED:**  Always read the pathwright instructions before starting the project to avoid the need fo sending another code review, according to the pathwright you are to use `Git Flow` not `GitHub Flow`.
Short Explanation Hint: 
`GitHub Flow`: Pull Requests is raise from `feature branches to the master`.
`Git Flow`: Pull Requests is raise from `feature branches to the development`.

## Congratulation :+1:  
_Go ahead and merge to master and don't forget to submit project completion form.
Goodluck in your next project :slightly_smiling_face:_

**Pull request merged**